packages/fxa-admin-panel/src/components/PageAccountSearch/Account/index.tsx

```diff
@@ -0,0 +1,389 @@
+              />
+            )}
+          </>
+          <>
+            {disabledAt != null && (
+              <TableRowYHeader
+                header="Disabled At"
+                className="bg-yellow-100"
+                value={`${disabledAtDate} (${disabledAt})`}
+                testId="account-disabled-at"
+              />
+            )}
+          </>
+        </TableYHeaders>
+
+        <TableXHeaders
+          header="Primary Email"
+          rowHeaders={['Email', 'Confirmed']}
+        >
+          <TableRowXHeader>
+            <span
+              data-testid="primary-email"
+              className={highlight(primaryEmail.email)}
+            >
+              {primaryEmail.email}
+            </span>
+
+            <ResultBoolean
+              isTruthy={primaryEmail.isVerified}
+              testId="primary-verified"
+            />
+          </TableRowXHeader>
+        </TableXHeaders>
+
+        <h3 className="header-lg">Secondary Emails</h3>
+        {secondaryEmails.length > 0 ? (
+          <TableXHeaders
+            rowHeaders={['Email', 'Confirmed']}
+            testId="secondary-section"
+          >
+            {secondaryEmails.map((secondaryEmail) => (
+              <TableRowXHeader key={secondaryEmail.createdAt}>
+                <span
+                  data-testid="secondary-email"
+                  className={highlight(secondaryEmail.email)}
+                >
+                  {secondaryEmail.email}
+                </span>
+                <ResultBoolean
+                  isTruthy={secondaryEmail.isVerified}
+                  testId="secondary-verified"
+                />
+              </TableRowXHeader>
+            ))}
+          </TableXHeaders>
+        ) : (
+          <p className="result-none">
+            This account doesn't have any secondary emails.
+          </p>
+        )}
+
+        <EmailBounces {...{ emailBounces, uid, emails, onCleared }} />
+
+        <h3 className="header-lg">
+          2FA / TOTP (Time-Based One-Time Passwords)
+        </h3>
+        {totps && totps.length > 0 ? (
+          <TableXHeaders rowHeaders={['Created At', 'Enabled', 'Confirmed']}>
+            {totps.map((totp: TotpType) => (
+              <TableRowXHeader key={totp.createdAt}>
+                <td data-testid="totp-created-at">
+                  {getFormattedDate(totp.createdAt)}
+                </td>
+                <ResultBoolean isTruthy={totp.enabled} testId="totp-enabled" />
+                <ResultBoolean
+                  isTruthy={totp.verified}
+                  testId="totp-verified"
+                />
+              </TableRowXHeader>
+            ))}
+          </TableXHeaders>
+        ) : (
+          <p className="result-none">
+            This account hasn't started 2FA / TOTP setup.
+          </p>
+        )}
+
+        <h3 className="header-lg">Account Recovery Key</h3>
+        {recoveryKeys && recoveryKeys.length > 0 ? (
```

Comment:
@dschom I didn't dig into the BE but I wonder if we got these confused with recovery codes because I don't know:
* why this is an array? If a user removes their recovery key, it no longer shows up here, so it seems like there should only be one 
* how users can "confirm" their recovery key? Seems like they just create it. I can't seem to "confirm" from Settings. I made this row a Y-header table since with "created at" there's possibly 4 pieces of data to display.

Reply:
I just double checked this. The behavior is correct and these are in fact account recovery keys. You are right that there's no reason it needs to be an array, but this doesn't necessarily seem _wrong_ imo.

When recovery keys are created. An email is sent to the user, and then they have to confirm this. So perhaps a better wording would be Email Confirmed.

Reply:
@dschom I just tried it and when I created a new recovery key, I received [this email ](https://storage.googleapis.com/mozilla-storybooks-fxa/commits/latest/fxa-auth-server/index.html?path=/story/fxa-emails-templates-postaddaccountrecovery--post-add-account-recovery) which doesn't ask for confirmation. It looks like the key is just automatically confirmed? 🤔